packages/salesforcedx-vscode-core/src/commands/util/sfdxCommandlet.ts

```diff
@@ -125,6 +168,77 @@ export abstract class SfdxCommandletExecutor<T>
     return;
   }
 
+  public outputResultPull(parser: ForcePullResultParser) {
+    const table = new Table();
+    const titleType = 'pull';
+
+    const successes: any = parser.getSuccesses();
+    const errors = parser.getErrors();
+    const pulledSource = successes ? successes.result.pulledSource : undefined;
+    if (pulledSource) {
+      const rows = pulledSource || (errors && errors.result);
+      const tableTitle = nls.localize(`table_title_${titleType}ed_source`);
+      const outputTable = this.getOutputTable(table, rows, tableTitle);
+
+      channelService.appendLine(outputTable);
+      if (pulledSource && pulledSource.length === 0) {
+        const noResults = nls.localize('table_no_results_found') + '\n';
+        channelService.appendLine(noResults);
+      }
+    }
+
+    if (errors) {
+      const { name, message, result } = errors;
+      if (result) {
+        const outputTable = this.getErrorTable(table, result, titleType);
+        channelService.appendLine(outputTable);
+      } else if (name && message) {
+        channelService.appendLine(`${name}: ${message}\n`);
+      }
+    }
+  }
+
+  protected getOutputTable(
+    table: Table,
+    rows: unknown,
+    outputTableTitle: string | undefined
+  ) {
+    const outputTable = table.createTable(
+      (rows as unknown) as Row[],
```

Comment:
weird. Is there a reason that `Row[]` isn't just the type of the parameter?

Reply:
When I try I get this:

<img width="700" alt="2023-05-19_17-55-43" src="https://github.com/forcedotcom/salesforcedx-vscode/assets/46458081/10d52be9-bd09-4259-8487-cee89da34c26">